file/writable_file_writer.cc

```diff
@@ -100,7 +104,104 @@ IOStatus WritableFileWriter::Append(const Slice& data) {
   } else {
     // Writing directly to file bypassing the buffer
     assert(buf_.CurrentSize() == 0);
-    s = WriteBuffered(src, left);
+    if (perform_data_verification_ && buffered_data_with_checksum_) {
+      buffered_data_crc32c_checksum_ = crc32c::Value(src, left);
+      s = WriteBufferedWithChecksum(src, left);
+    } else {
+      s = WriteBuffered(src, left);
+    }
+  }
+
+  TEST_KILL_RANDOM("WritableFileWriter::Append:1");
+  if (s.ok()) {
+    filesize_ += data.size();
+  }
+  return s;
+}
+
+IOStatus WritableFileWriter::Append(const Slice& data,
```

Comment:
Can we reuse code from ```WritableFileWriter::Append```, since most of the logic is the same?

Reply:
I will try to combine in the same Append with default checksum = 0.